kustomization/components/samba-upload/deployment.yml

```diff
@@ -0,0 +1,54 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: samba
+spec:
+  revisionHistoryLimit: 4
+  strategy:
+    type: Recreate
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: samba
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: samba
+    spec:
+      affinity:
+        podAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            - topologyKey: kubernetes.io/hostname
+              labelSelector:
+                matchExpressions:
+                  - key: app.kubernetes.io/name
+                    operator: In
+                    values:
+                      - paperless
+      volumes:
+        - name: paperless-scans
+          persistentVolumeClaim:
+            claimName: paperless-scans
+      containers:
+        - name: samba
+          image: ghcr.io/servercontainers/samba:smbd-only-latest
+          ports:
+            - containerPort: 445
+            - containerPort: 139
+          env:
+            - name: ACCOUNT_hp
+              valueFrom:
+                secretKeyRef:
+                  name: paperless-secrets
+                  key: hp_samba_password
```

Comment:
This secret isn't documented.  In an ideal world, we'd do this more like the mosquitto component that will just take all the users/passwords in a secret and configure them, but we can do that later.

I would prefer if this was a bit more generic than `hp` (like `printer`), but I have no idea if that will break out.

Reply:
Oh hah, this is something I should have abstracted out but somehow missed entirely